src/runtime-rs/crates/agent/src/sock/vsock.rs

```diff
@@ -26,7 +33,50 @@ impl Vsock {
 
 #[async_trait]
 impl Sock for Vsock {
-    async fn connect(&self, _config: &ConnectConfig) -> Result<Stream> {
-        todo!()
+    async fn connect(&self, config: &ConnectConfig) -> Result<Stream> {
+        let retry_times = config.reconnect_timeout_ms / config.dial_timeout_ms;
```

Comment:
Please add a check to ensure `config.dial_timeout_ms > 0` to avoid a crash here.

Reply:
I check it in config validation for both vsock and hvsock.